js/crosswords.js

```diff
@@ -1113,54 +1140,49 @@ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
         this.hidden_input.on('keydown', $.proxy(this.keyPressed, this));
       }
 
-      handleClickOpenMenu(event) {
-        const menuContainer = $(event.target).closest('.cw-menu-container');
-        const menu = menuContainer.find('.cw-menu');
-        if (menu.is(':visible')) menu.hide();
-        else menu.show();
+      handleClickWindow(event) {
+        this.root.find('.cw-menu').hide();
       }
 
-      handleMenuBlur(event) {
+      handleClickOpenMenu(event) {
         const menuContainer = $(event.target).closest('.cw-menu-container');
         const menu = menuContainer.find('.cw-menu');
-        const focusTarget = event.relatedTarget;
-        if (!focusTarget || !$.contains(menu.get(0), focusTarget)) {
-          menu.hide();
+        if (!menu.is(':visible')) {
+          // This is horrible: delay opening the menu so that
+          // `handleClickWindow` can run first and close all previously-open
+          // menus
+          setTimeout(() => {
+            menu.show();
+          });
         }
       }
 
-      handleClickMenuButton(event) {
-        const menuContainer = $(event.target).closest('.cw-menu-container');
-        const menu = menuContainer.find('.cw-menu');
-        menu.hide();
-      }
-
       // Create a generic modal box with content
       createModalBox(title, content, button_text = 'Close') {
         // Set the contents of the modal box
         const modalContent = `
         <div class="modal-content">
           <div class="modal-header">
-            <span class="close" id="modalClose">&times;</span>
+            <span class="modal-close">&times;</span>
             <span class="modal-title">${title}</span>
           </div>
           <div class="modal-body">
             ${content}
           </div>
           <div class="modal-footer">
-            <button id="modal-button" class="modal-button">${button_text}</button>
+            <button class="cw-button">${button_text}</button>
```

Comment:
```suggestion
            <button class="cw-button" id="modal-button">${button_text}</button>
```
unless this is a bad idea for some reason?

Reply:
No reason, thanks! IDs kind of suck but it’s really not a big deal in this case, just habit